### DISCUSSION TITLE
How to generate a front view of a 3D model？

### DISCUSSION THREAD

**Discussion by hongtaoliuxing** [neperfepx/neper#819] [Q&A] [answered]
Hello everyone.

I need the front view and three views of the model. I tried using the command below, but I couldn't debug the front view. Are there any other commands?

neper -T -n 100 -transform "rotate(0,0,1,30),rotate(1,0,0,-15),rotate(0,1,0,-15)"

Thanks.

**Comment by rquey** [comment, marked as answer]
Move the camera, not the tessellation, see [-cameracoo](https://neper.info/doc/neper_v.html#cmdoption-cameracoo).

**Reply by hongtaoliuxing** [reply]
Thank you for your suggestion. I tried this command, but it didn't seem to work quite right. I'm not sure if it's correct.

neper -V n100-id1.tess -cameracoo 1:-10:1 -datacellcol id -print 100

**Reply by rquey** [reply]
Assuming you built a 1 x 1 x 1 tessellation, x=1 and z=1 do not exactly correspond to front and top views.
Use `-cameracoo x:-10:z` instead. You may be also interested in `-cameraprojection orthographic` (for orthographic projection instead of perspective projection) and `-camerasky` to set the top direction of the image.

```
neper -T -n 100
neper -V n100-id1.tess -print xyzview
neper -V n100-id1.tess -cameracoo x:-5:z -cameraprojection orthographic -camerasky 0:0:1 -print yview
neper -V n100-id1.tess -cameracoo x:y:-5 -cameraprojection orthographic -camerasky 1:0:0 -print zview
neper -V n100-id1.tess -cameracoo -5:y:z -cameraprojection orthographic -camerasky 0:1:0 -print xview
```
![xyzview](https://github.com/neperfepx/neper/assets/2659376/7f7b9e11-e7fa-497a-ae24-030e6d7f6b83)
![yview](https://github.com/neperfepx/neper/assets/2659376/ff5aa6a8-972b-4b7e-baaa-4ffaf45f3e68)
![zview](https://github.com/neperfepx/neper/assets/2659376/2595b8b5-5cf0-4235-bb0a-e8dc8b43d53b)
![xview](https://github.com/neperfepx/neper/assets/2659376/e1f4f396-1f5c-4ed9-86fb-8a09cdeb2098)

**Reply by hongtaoliuxing** [reply]
Thank you again for your advice, have a pleasant life!
